### PR TITLE
Return User Info from Login Endpoint

### LINKED ISSUE #123: Return User Details from Login Endpoint
## Overview

Currently the login endpoint doesn't return any data:

https://github.com/azavea/iow-boundary-tool/blob/100e8d049fa95d43e05ff63744f1ef9acb23332a/src/django/api/views/auth.py#L35-L37

Change this so that this endpoint returns:

- [x] The user's email
- [x] The user's role
- [x] The list of utilities the user has access to
  - For validators, this can be an empty list
  - For contributors, this should have at least one value for the app to function correctly

### PR DESCRIPTION
**Pull request by rajadain**
## Overview

Returns user details from the login endpoint. Both when logging in a new user, and when fetching an already logged in user, we return details about the user, which include their email, Role, and list of Utilities.

Adds a Utility and State serializer along the way. Updates front-end redux arrangement to consume the new user object for login state management. Also uses the user's list of utilities to correctly render the Nav Bar.

Closes #123 

### Demo

image

## Testing Instructions

- Checkout this branch and `server`
- Go to http://localhost:4545/ and log in as a contributor
  - [x] Ensure you are logged in correctly
- Go to http://localhost:4545/draw and http://localhost:4545/submissions
  - [x] Ensure the user's utility is shown correctly in the nav bar
- Log out, and log back in
  - [x] Ensure logging out works correctly
  - [x] Ensure logging in returns you to the previous page instead of taking you to Welcome

## Checklist

- [x] `fixup!` commits have been squashed
- [x] `CHANGELOG.md` updated with summary of features or fixes, following [Keep a Changelog](https://keepachangelog.com/en/1.0.0/) guidelines
- [ ] `README.md` updated if necessary to reflect the changes
- [ ] CI passes after rebase

### CODE REVIEW

**Comment by jacobtylerwalls** [comment]
Taking a look now.

**Comment by rajadain** [comment]
For some reason the linter integration with my VSCode isn't working, leading to all the linting errors. [9552480](https://github.com/azavea/iow-boundary-tool/pull/136/commits/9552480281ed82734e95a2b8139b00d6545c714e) should be the last fix that takes care of it.

**Comment by jacobtylerwalls** [comment]
One thing quickly, since I know you're online. With this PR we now lose this ability to keep the user logged in after refresh, i.e. when we lose redux state. I don't know if our usual approach is to persist the redux state, or to tolerate more API calls, or to just tolerate logging the user out.

**Comment by rajadain** [comment]
Nice catch! Had missed saving the user state in PrivateRoute. Fixed in [33930fe](https://github.com/azavea/iow-boundary-tool/pull/136/commits/33930fe266fac244668bea84fd69e84e71ba8728)

**Review by jacobtylerwalls — APPROVED** [review]
Tested, works great!

File: src/django/api/serializers/state.py [hunk 1]
```diff
@@ -0,0 +1,10 @@
+from sre_parse import State
```
[the remark this answers is not in the document]
Comment (jacobtylerwalls) [reply src/django/api/serializers/state.py]: Oh, was this just VS Code being "helpful"? 😆

[the remark this answers is not in the document]
Comment (rajadain) [reply src/django/api/serializers/state.py]: Haha yes 😅

**Comment by rajadain** [comment]
Thanks for reviewing!

**Pull request merged**